pkg/api/validation/validation.go

```diff
@@ -441,21 +441,25 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource) errs.Valid
 
 func validateGitRepoVolumeSource(gitRepo *api.GitRepoVolumeSource) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if gitRepo.Repository == "" {
+
+	if len(gitRepo.Repository) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("repository"))
 	}
+
+	pathErrs := validateVolumeSourcePath(gitRepo.Directory, "directory")
+	allErrs = append(allErrs, pathErrs...)
 	return allErrs
 }
 
 func validateISCSIVolumeSource(iscsi *api.ISCSIVolumeSource) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if iscsi.TargetPortal == "" {
+	if len(iscsi.TargetPortal) == 0 {
```

Comment:
@resouer I had meant, can we please strink these changes to string tests from this PR and fix them in another PR?  It makes this PR about more than one thing and makes it tough to see the real changes to this file.